csrc/id_model/indexing.cpp

```diff
@@ -303,6 +312,112 @@ class AllocationDomainSetup : private kir::IrVisitor {
     return IndexingAllocationInfo{actual_allocation_domains, actual_strides};
   }
 
+  // Reorder non-logical allocation domains to follow the ordering of
+  // the logical domain. This is necessary when an allocation domain
+  // includes a vectorized loop iter domain since it must be at the
+  // innermost position but that may not be the case in the loop
+  // domain. Not strictly necessary otherwise, but this should also
```

Comment:
If a non-innermost loop domain is vectorized is that treated as innermost for inlining? That is, it cannot inline further than the vectorized domain right? I wonder in what cases we would want to have a vectorized loop iterdomain that is not inner-most.

Reply:
> it cannot inline further than the vectorized domain right?

Right, I forgot that effect to inlining. So, if a non-innermost loop domain is vectorized, it's still vectorized properly, and the maximum inlining position is limited by the position of the vectorized domain. If we changed the scheduler to always place vectorized domains at the innermost loop domain position, it would also change the inlining positions. I'm not sure if that would have any performance implication in practice.

> I wonder in what cases we would want to have a vectorized loop iterdomain that is not inner-most.

I don't know. Maybe that's just for simplicity of scheduling?